elements/pf-checkbox/pf-checkbox.ts

```diff
@@ -0,0 +1,180 @@
+import { LitElement, html, type PropertyValues } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+
+import styles from './pf-checkbox.css';
+import { property } from 'lit/decorators/property.js';
+import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
+import { SlotController } from '@patternfly/pfe-core/controllers/slot-controller.js';
+
+export class RestoreEvent extends Event {
+  constructor(public element: PfCheckbox) {
+    super('restore', { bubbles: true, composed: true });
+  }
+}
+
+/**
+ * A **checkbox** is used to select a single item or multiple items, typically to choose elements to perform an action or to reflect a binary setting.
+ *
+ * @slot - Place nested (i.e. 'controlled') patternfly form-control elements here
+ * @slot description - Description text
+ * @slot body - Body text
+ * @fires {RestoreEvent} restore - when the form state is restored.
+ * @cssprop --pf-c-check--GridGap {@default var(--pf-global--spacer--xs, 0.25rem) var(--pf-global--spacer--sm, 0.5rem)}
+ * @cssprop --pf-c-check__label--disabled--Color {@default var(--pf-global--disabled-color--100, #6a6e73)}
+ * @cssprop --pf-c-check__label--Color {@default var(--pf-global--Color--100, #151515)}
+ * @cssprop --pf-c-check__label--FontWeight {@default var(--pf-global--FontWeight--normal, 400)}
+ * @cssprop --pf-c-check__label--FontSize {@default var(--pf-global--FontSize--md, 1rem)}
+ * @cssprop --pf-c-check__label--LineHeight {@default var(--pf-global--LineHeight--sm, 1.3)}
+ * @cssprop --pf-c-check__input--Height {@default var(--pf-c-check__label--FontSize)}
+ * @cssprop --pf-c-check__input--MarginTop {@default calc(((var(--pf-c-check__label--FontSize) * var(--pf-c-check__label--LineHeight)) - var(--pf-c-check__input--Height)) / 2)}
+ * @cssprop --pf-c-check__description--FontSize {@default var(--pf-global--FontSize--sm, 0.875rem)}
+ * @cssprop --pf-c-check__description--Color {@default var(--pf-global--Color--200, #6a6e73)}
+ * @cssprop --pf-c-check__body--MarginTop {@default var(--pf-global--spacer--sm, 0.5rem)}
+ * @cssprop --pf-c-check__label-required--MarginLeft {@default var(--pf-global--spacer--xs, 0.25rem)}
+ * @cssprop --pf-c-check__label-required--FontSize {@default var(--pf-global--FontSize--sm, 0.875rem)}
+ * @cssprop --pf-c-check__label-required--Color {@default var(--pf-global--danger-color--100)}
+ */
+@customElement('pf-checkbox')
+export class PfCheckbox extends LitElement {
+  static readonly styles = [styles];
+
+  static formAssociated = true;
+
+  static shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  @property() label?: string;
+
+  @property() value = '';
+
+  @property({ type: Boolean, reflect: true }) indeterminate = false;
```

Comment:
Should these have jsdoc descriptions?  

Edit: Maybe not since they're based on native checkbox attributes.